src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userOptional = userService.findByEmail(email);
+        if (userOptional.isEmpty()) {
+            throw new AuthenticationException("Can't authenticate user");
+        }
+        User user = userOptional.get();
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (hashedPassword.equals(user.getPassword())) {
+            return user;
+        }
+        throw new AuthenticationException("Can't authenticate user");
```

Comment:
checklist, try making only one throw statement, instead of isEmpty use isPresent && passwordCheck, if fail -> exception

Reply:
fixed